Sources/Foundation/NSError.swift

```diff
@@ -300,7 +300,7 @@ public protocol CustomNSError : Error {
 public extension CustomNSError {
     /// Default domain of the error.
     static var errorDomain: String {
-        return String(reflecting: self)
+        return _typeName(type(of: self), qualified: true)
```

Comment:
@jmschonfeld you've used this function for `Predicate` - does it seem right here too?

Reply:
I've got it little bit wrong. Because it is static var, there is no need to call `type(of:)`, it should be `_typeName(self, qualified: true)`
<img width="394" alt="Screenshot 2024-11-04 at 21 14 25" src="https://github.com/user-attachments/assets/11789749-3bab-494c-80e0-cc77b12a52d6">

Reply:
Yeah `_typeName(self)` is a fine way to get the fully qualified name here - just to confirm, @Azoy is it always the case that `String(reflecting: someType)` will always just return the fully qualified type name here or do you know of any types that might somehow produce a different value via `String(reflecting:)` here (just to make sure we don't accidentally change behavior)?